File: backend-java/src/main/java/com/example/pricing/entity/AgentRunLog.java
package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * Agent 运行日志实体，对应多智能体在任务中的发言、证据和结论记录。
 */
@Data
@TableName("agent_run_log")
public class AgentRunLog {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("task_id")
    private Long taskId;

    @TableField("role_name")
    private String roleName;

    @TableField("speak_order")
    private Integer speakOrder;

    @TableField("thought_content")
    private String thoughtContent;

    @TableField("thinking_summary")
    private String thinkingSummary;

    @TableField("evidence_json")
    private String evidenceJson;

    @TableField("suggestion_json")
    private String suggestionJson;

    @TableField("final_reason")
    private String finalReason;

    @TableField("display_order")
    private Integer displayOrder;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: backend-python/app/models/agent_run_log.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Integer, JSON, String, Text, text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class AgentRunLog(Base):
    __tablename__ = "agent_run_log"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    task_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    role_name: Mapped[str] = mapped_column(String(50), nullable=False)
    speak_order: Mapped[int] = mapped_column(Integer, nullable=False)
    thought_content: Mapped[str | None] = mapped_column(Text, nullable=True)
    thinking_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    evidence_json: Mapped[list[dict] | None] = mapped_column(JSON, nullable=True)
    suggestion_json: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    raw_output_json: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    final_reason: Mapped[str | None] = mapped_column(Text, nullable=True)
    display_order: Mapped[int | None] = mapped_column(Integer, nullable=True)
    stage: Mapped[str] = mapped_column(String(20), nullable=False, server_default=text("'completed'"))
    run_attempt: Mapped[int] = mapped_column(Integer, nullable=False, server_default=text("0"))
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"))
